server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/ImapRequestFrameDecoder.java

```diff
@@ -221,17 +221,21 @@ public void close() throws IOException {
 
     @Override
     protected synchronized ChannelBuffer newCumulationBuffer(ChannelHandlerContext ctx, int minimumCapacity) {
-        @SuppressWarnings("unchecked")
         Map<String, Object> attachment = (Map<String, Object>) ctx.getAttachment();
-        int size = (Integer) attachment.get(NEEDED_DATA);
-        
-        if (inMemorySizeLimit > 0) {
-            return ChannelBuffers.dynamicBuffer(Math.min(size, inMemorySizeLimit), ctx.getChannel().getConfig().getBufferFactory());
-        } else {
+        if (attachment.containsKey(NEEDED_DATA)) {
+            @SuppressWarnings("unchecked")
+            int size = (Integer) attachment.get(NEEDED_DATA);
```

Comment:
is the map mutable? Can the result be different between the `contains` and the `get` call?